src/crypto.ts

```diff
@@ -154,6 +152,11 @@ export class Crypto extends AsyncOptionalCreatable<CryptoOptions> {
         dec = decipher.update(secret, 'hex', 'utf8');
         dec += decipher.final('utf8');
       } catch (e) {
+        const useGenericUnixKeychain =
+          env.getBoolean('SFDX_USE_GENERIC_UNIX_KEYCHAIN') || env.getBoolean('USE_GENERIC_UNIX_KEYCHAIN');
+        if (os.platform() === 'darwin' && !useGenericUnixKeychain) {
+          e.message += Messages.loadMessages('@salesforce/core', 'crypto').getMessage('MacKeychainOutOfSync');
```

Comment:
What does the error look like with the `+=`? It seems a little weird to have "<random crypto error>. We have encountered...". It might be worth it to just overwrite it, put it on a new line, put it in an error action, or something else to make it a little more readable. Or change the verbiage. Like "<random crypto error>. This is usually caused by...".

Reply:
IF the message is added : `Failed to decipher auth data. reason: Unsupported state or unable to authenticate data
We’ve encountered an error with the Mac keychain being out of sync with your 'sfdx' credentials. To fix the problem, sync your credentials by logging into your Dev Hub.` vs `Failed to decipher auth data. reason: Unsupported state or unable to authenticate data`

Reply:
Ya, that seems weird to have the We've right after data with no punctuation or anything. 

```
Failed to decipher auth data. reason: Unsupported state or unable to authenticate data 

Try this: The Mac keychain might be out of sync with your 'sfdx' credentials. To fix the problem, sync your credentials by logging into your Dev Hub using an auth command.
```

Note, you get the `Try this` by adding an action to the error. That to me looks way more clear. I would run it by Terresa one more time.